ch_pipeline/analysis/timing.py

```diff
@@ -83,6 +119,33 @@ def process(self, tstream):
         # If requested, extract the input flags
         input_flags = tstream.input_flags[:] if self.use_input_flags else None
 
+        # Check needs_timing_correction flags time ranges to see if input timestream
+        # falls within the interval
+        needs_timing_correction = False
+        for flag in self.flags:
+            if timestamp[0] >= flag.start_time and timestamp[0] <= flag.finish_time:
```

Comment:
You may want to check for the case that the timing correction needed status may change within a file. Not sure what to do in that case (probably reject it), but whatever you do a clear logging statement would be good.

Reply:
I am not sure what you mean by reject it? Does it mean not apply the timing correction? That is what I did for now!